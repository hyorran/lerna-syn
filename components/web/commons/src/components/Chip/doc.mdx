---
name: Chip
menu: Commons
---

import { Playground, PropsTable } from 'docz'
import Chip, { ComponentWithProps } from './'

# Chip

A Simple Chip Example

## Basic Usage
<Playground>
  <Chip
    label="Name of field"
    value="my value"
    tooltip="My message!"
  />
</Playground>

## Props

<PropsTable of={ ComponentWithProps } />
