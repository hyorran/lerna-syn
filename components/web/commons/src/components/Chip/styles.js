import Colors from '@syntesis/c-styles/src/styles/Colors'
import Fonts from '@syntesis/c-styles/src/styles/Fonts'

export default () => ({
  chip: {
    display: 'flex',
    flex: 1,
    margin: 3
  },
  chipWithTooltip: {
    cursor: 'help'
  },
  tooltip: {
    backgroundColor: Colors.text,
    color: Colors.white,
    fontSize: Fonts.fontSize.SS,
    border: `1px solid ${ Colors.text }`
  },
  label: {
    width: '100%',
    marginTop: -1
  },
  labelContainer: {
    display: 'inline-flex',
    flex: 1,
    width: '100%',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: [0, 10]
  },
  labelValue: {
    paddingLeft: 15
  }
})
