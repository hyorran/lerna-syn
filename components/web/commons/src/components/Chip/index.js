import React from 'react'
import PropTypes from 'prop-types'
import isEmpty from 'lodash/isEmpty'
import { withStyles } from '@material-ui/core/styles'
import ChipMui from '@material-ui/core/Chip'
import Typography from '@material-ui/core/Typography'
import Tooltip from '@material-ui/core/Tooltip'
import OpenInNewIcon from '@syntesis/c-icons/src/OpenInNewIcon'

import styles from './styles'

const Chip = (props) => {
  const {
    classes,
    label,
    value,
    tooltip,
    externalLinkUrl,
    textAsError,
    ...otherProps
  } = props

  const hasTooltip = !isEmpty(tooltip)

  let chipProps = {}

  if (!isEmpty(externalLinkUrl)) {
    chipProps = {
      ...chipProps,
      clickable: true,
      onDelete: () => window.open(externalLinkUrl, '_blank'),
      deleteIcon: <OpenInNewIcon />
    }
  }

  let response = (
    <ChipMui
      label={ (
        <div className={ classes.labelContainer }>
          <div>
            <b>{ label }</b>:
          </div>
          <div className={ classes.labelValue }>
            {
              isEmpty(value)
                ? '-'
                : (
                  <Typography color={ textAsError ? 'secondary' : undefined }>
                    { value }
                  </Typography>
                )
            }
          </div>
        </div>
      ) }
      classes={ {
        label: classes.label
      } }
      className={ [classes.chip, hasTooltip ? classes.chipWithTooltip : null].join(' ') }
      { ...chipProps }
      { ...otherProps }
    />
  )

  if (hasTooltip) {
    response = (
      <Tooltip title={ tooltip } classes={ { tooltip: classes.tooltip } }>
        { response }
      </Tooltip>
    )
  }

  return response
}

Chip.propTypes = {
  /** Provided by material-ui. */
  classes: PropTypes.object.isRequired,
  /** Internal label of the component. */
  label: PropTypes.string.isRequired,
  /** Value inside of the component. */
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element
  ]),
  /** Text or element displayed inside the tooltip. */
  tooltip: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element
  ]),
  textAsError: PropTypes.bool,
  /** Can be used to display an link inside the component. */
  externalLinkUrl: PropTypes.string
}

Chip.defaultProps = {
  value: null,
  tooltip: null,
  textAsError: false
}

export default withStyles(styles)(Chip)

export { Chip as ComponentWithProps }
