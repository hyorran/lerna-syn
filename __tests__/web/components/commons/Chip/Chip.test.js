import React from 'react'
import Chip from '@syntesis/c-commons/src/components/Chip'
import { mount } from 'enzyme'

describe('<Chip />', () => {
  it('rendred', () => {
    const wrapper = mount((
      <Chip label="teste" />
    ))
    expect(wrapper).toMatchSnapshot()
  })
})
