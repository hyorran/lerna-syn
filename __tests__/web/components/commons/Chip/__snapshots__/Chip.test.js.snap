// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> rendred 1`] = `
<WithStyles(Chip)
  label="teste"
>
  <Chip
    classes={
      Object {
        "chip": "Chip-chip-1",
        "chipWithTooltip": "Chip-chipWithTooltip-2",
        "label": "Chip-label-4",
        "labelContainer": "Chip-labelContainer-5",
        "labelValue": "Chip-labelValue-6",
        "tooltip": "Chip-tooltip-3",
      }
    }
    label="teste"
    textAsError={false}
    tooltip={null}
    value={null}
  >
    <WithStyles(Chip)
      className="Chip-chip-1 "
      classes={
        Object {
          "label": "Chip-label-4",
        }
      }
      label={
        <div
          className="Chip-labelContainer-5"
        >
          <div>
            <b>
              teste
            </b>
            :
          </div>
          <div
            className="Chip-labelValue-6"
          >
            -
          </div>
        </div>
      }
    >
      <Chip
        className="Chip-chip-1 "
        classes={
          Object {
            "avatar": "MuiChip-avatar-19",
            "avatarChildren": "MuiChip-avatarChildren-22",
            "avatarColorPrimary": "MuiChip-avatarColorPrimary-20",
            "avatarColorSecondary": "MuiChip-avatarColorSecondary-21",
            "clickable": "MuiChip-clickable-10",
            "clickableColorPrimary": "MuiChip-clickableColorPrimary-11",
            "clickableColorSecondary": "MuiChip-clickableColorSecondary-12",
            "colorPrimary": "MuiChip-colorPrimary-8",
            "colorSecondary": "MuiChip-colorSecondary-9",
            "deletable": "MuiChip-deletable-13",
            "deletableColorPrimary": "MuiChip-deletableColorPrimary-14",
            "deletableColorSecondary": "MuiChip-deletableColorSecondary-15",
            "deleteIcon": "MuiChip-deleteIcon-27",
            "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary-28",
            "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary-29",
            "deleteIconOutlinedColorPrimary": "MuiChip-deleteIconOutlinedColorPrimary-30",
            "deleteIconOutlinedColorSecondary": "MuiChip-deleteIconOutlinedColorSecondary-31",
            "icon": "MuiChip-icon-23",
            "iconColorPrimary": "MuiChip-iconColorPrimary-24",
            "iconColorSecondary": "MuiChip-iconColorSecondary-25",
            "label": "MuiChip-label-26 Chip-label-4",
            "outlined": "MuiChip-outlined-16",
            "outlinedPrimary": "MuiChip-outlinedPrimary-17",
            "outlinedSecondary": "MuiChip-outlinedSecondary-18",
            "root": "MuiChip-root-7",
          }
        }
        color="default"
        component="div"
        label={
          <div
            className="Chip-labelContainer-5"
          >
            <div>
              <b>
                teste
              </b>
              :
            </div>
            <div
              className="Chip-labelValue-6"
            >
              -
            </div>
          </div>
        }
        variant="default"
      >
        <div
          className="MuiChip-root-7 Chip-chip-1 "
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiChip-label-26 Chip-label-4"
          >
            <div
              className="Chip-labelContainer-5"
            >
              <div>
                <b>
                  teste
                </b>
                :
              </div>
              <div
                className="Chip-labelValue-6"
              >
                -
              </div>
            </div>
          </span>
        </div>
      </Chip>
    </WithStyles(Chip)>
  </Chip>
</WithStyles(Chip)>
`;
